WSEvents.h

```diff
@@ -41,9 +40,18 @@ class WSEvents : public QObject {
     uint64_t GetRecordingTime();
     const char* GetRecordingTimecode();
 
+    bool Heartbeat_active;
+    bool Heartbeat_CurrentProfile_active;
+    bool Heartbeat_CurrentScene_active;
+    bool Heartbeat_Streaming_active;
+    bool Heartbeat_Recording_active;
+    bool Heartbeat_TotalStreamTime_active;
+    bool Heartbeat_TotalRecordTime_active;
```

Comment:
Except for `Heartbeat_active`, what is the purpose of manually enabling/disabling each feature of the heartbeat message?
From what I see in the event's implementation, it costs nothing to get all of these from the OBS API, and isn't much of an overhead in the WebSocket messages.

Reply:
> Why add/sub: I thought lets keep is small an neat. Also if in future extra key/value pairs are coded than it can grow without everybody needing the data. (me once a programmer of 8-bit micro controllers.)

I did & do embedded programming too (maybe not as experienced as you) and I understand why you've it designed it like this. Can't disagree much with that 😄
However, obs-websocket clients are mainly desktop computers, smartphones and tablets connected with the server over a high-capacity LAN/WLAN network, or through a broadband connection in the case of users taking that risk. Over those links, there's enough capacity to carry this message at its default rate or even several times per second.
(Furthermore, `StreamStatus` doesn't have selectable fields and it didn't bother anyone)

I'm sorry to say this, especially given this is your first contribution (welcome!), that you've spent time on this and that it's been waiting for too long, but selectable fields for the `Heartbeat` event aren't something I'd like to keep in obs-websocket's protocol.
However, you can keep these on your own version if you want to use it.

Nevertheless, this is the last thing to fix before I can merge this into `master`, hence including in obs-websocket. Your code looks good to me so far. Great work!

Reply:
Thanks for your feedback. Up will limit the request to on "on/off" only as per your earlier comment, and that the event always sends everything in "on" situation, and nothing in "off" situation.
So far it has been a good experience to dive into OBS.
By the way... congrats with releasing 4.2.0!